ui/src/criteria/attribute.tsx

```diff
@@ -81,17 +129,115 @@ function AttributeEdit(props: AttributeEditProps) {
   const hintDisplayName = (hint: tanagra.EnumHintValue) =>
     hint.displayName || "Unknown Value";
 
-  const enumHintValues = underlay.entities
+  const attributeFilterHint = underlay.entities
     .find((g) => g.name === underlay.primaryEntity)
-    ?.attributes?.find((attribute) => attribute.name === props.data.attribute)
-    ?.attributeFilterHint?.enumHint?.enumHintValues;
+    ?.attributes?.find(
+      (attribute) => attribute.name === props.data.attribute
+    )?.attributeFilterHint;
+
+  const year = new Date().getFullYear();
+
+  if (isValid(attributeFilterHint?.integerBoundsHint)) {
+    const integerBoundsHint = attributeFilterHint?.integerBoundsHint;
+
+    if (
+      integerBoundsHint?.max !== null &&
+      integerBoundsHint?.max !== undefined &&
+      integerBoundsHint?.min !== null &&
+      integerBoundsHint?.min !== undefined
+    ) {
+      const min = year - integerBoundsHint?.max;
+      const max = year - integerBoundsHint.min;
+
+      const [minValue, setMinValue] = React.useState<number>(min);
+      const [maxValue, setMaxValue] = React.useState<number>(max);
+
+      const updateValue = (newMin: number, newMax: number) => {
+        setMinValue(newMin);
+        setMaxValue(newMax);
+        props.dispatchFn(
+          produce(props.data, (data) => {
+            data.min = newMin;
+            data.max = newMax;
+          })
+        );
+      };
+
+      React.useEffect(() => updateValue(min, max), []);
+
+      const handleChange = (event: Event, newValue: number | number[]) => {
+        newValue = newValue as number[];
+        const [newMin, newMax] = newValue;
+        updateValue(newMin, newMax);
+      };
+
+      const handleInputChange = (
+        event: React.ChangeEvent<HTMLInputElement>
+      ) => {
+        const newMin =
+          event.target.value === "" ? min : Number(event.target.value);
+        const newMax =
+          event.target.value === "" ? max : Number(event.target.value);
+        updateValue(Math.min(newMin, minValue), Math.max(newMax, maxValue));
```

Comment:
What is the purpose on Math.min/.max here? Doesn't that it prevent min from ever increasing and max from ever decreasing?

Reply:
I reason why I do that is there is 2 pointers. And mostly, left represents min, and right represent max. But if left goes over to the right, and then the right will be left(min), and left will be right(max). it will reverse. 
in order to avoid this corner case, I just write this

Reply:
Okay, I see what the goal is now. So what happens if min is 20, max is 30, and you start typing 40 into the max input overwriting the 30? When you enter the 4, doesn't that change the minimum to 4 and the maximum to 10? Then adding the 0 for 40 means the minimum is 4 and the maximum is 40? And the maximum input change to 10 while you're typing in it?

Beyond the issues when editing the inputs, having the inputs cross over each other feels weird to me. It's functional but unexpected and the current AoU slider doesn't allow the limits to cross either. We should emulate that behavior here.

Reply:
I know how to solve the problem for over sliding. There is a parameter disableSwap. 
I checked it will not change in the middle of the typing.

Reply:
I meditated on the input behavior for a while. I update the handleInputChange behavior and now it match the AoU behavior exactly.